soccer/Robot.cpp

```diff
@@ -265,10 +268,16 @@ float OurRobot::kickTimer() const {
 }
 
 void OurRobot::dribble(uint8_t speed) {
-    uint8_t scaled = *config->dribbler.multiplier * speed;
-    control->set_dvelocity(scaled);
+    Geometry2d::Rect modifiedField = Field_Dimensions::Current_Dimensions.FieldRect();
+    modifiedField.expand(*new Point(*_dribbleOutOfBoundsOffset,*_dribbleOutOfBoundsOffset));
```

Comment:
For future reference, you are are trying to pass a object by value to a function, just do `modifiedField.expand(Point(...))`. Here you are creating an object on the heap then instantly throwing away the resulting address of that object on the heap which causing a memory leak. Every `new` needs a corresponding delete (in most cases)

Reply:
Thanks Joe. I should of known about the memory leak. Though in regards to the second comment I asked Woodward if that would work.